src/CanvasRender/index.js

```diff
@@ -2,12 +2,17 @@
 
 import * as React from 'react'
 import PropTypes from 'prop-types'
+import {some} from 'lodash'
 
 import DEFAULT_THEME from '../defaultTheme'
 import {WIDTH, HEIGHT} from '../chartCore/defaults'
 
 import basicRender from './basicRender'
 
+function dependenciesHaveChanged(dependencies, props, prevProps) {
+  return some(dependencies, dependency => props[dependency] !== prevProps[dependency])
```

Comment:
This won't matter if we go with the `PureComponent` option below, but you can use `Array.prototype.some` here (`dependencies.some(...)`)